src/app/rpmostree-compose-builtin-commit2jigdo.c

```diff
@@ -0,0 +1,1154 @@
+
+        /* Build up the variant from sorted data */
+        g_autoptr(GVariantBuilder) objid_xattr_builder = g_variant_builder_new ((GVariantType*)"a(su)");
+        for (guint i = 0; i < objidxattrs->len; i++)
+          {
+            GVariant *objidxattr = objidxattrs->pdata[i];
+            g_variant_builder_add (objid_xattr_builder, "@(su)", objidxattr);
+          }
+        g_autoptr(GVariant) objid_xattrs_final = g_variant_ref_sink (g_variant_builder_end (objid_xattr_builder));
+
+        g_autofree char *path = g_strconcat (RPMOSTREE_JIGDO_XATTRS_PKG_DIR, "/", nevra, NULL);
+        if (!glnx_file_replace_contents_at (oirpm_tmpd.fd, path, g_variant_get_data (objid_xattrs_final),
+                                            g_variant_get_size (objid_xattrs_final),
+                                            GLNX_FILE_REPLACE_NODATASYNC,
+                                            cancellable, error))
+          return FALSE;
+
+        g_hash_table_iter_remove (&it);
+      }
+  }
+
+  if (!opt_only_contentdir)
+    {
+      const char *commit_version;
+      if (!g_variant_lookup (commit_inline_meta, OSTREE_COMMIT_META_KEY_VERSION, "&s", &commit_version))
+        commit_version = NULL;
+
+      g_autoptr(GPtrArray) rpmbuild_argv = g_ptr_array_new ();
+      g_ptr_array_add (rpmbuild_argv, (char*)"rpmbuild");
+      g_ptr_array_add (rpmbuild_argv, (char*)"-bb");
+      /* We use --build-in-place to avoid having to compress the data again into a
+       * Source only to immediately uncompress it.
+       */
+      g_ptr_array_add (rpmbuild_argv, (char*)"--build-in-place");
+      g_autofree char *ostree_version_macro_arg = NULL;
+      if (commit_version)
+        {
+          g_ptr_array_add (rpmbuild_argv, (char*)"-D");
+          ostree_version_macro_arg = g_strconcat ("ostree_version ", commit_version, NULL);
+          g_ptr_array_add (rpmbuild_argv, ostree_version_macro_arg);
+        }
+      g_ptr_array_add (rpmbuild_argv, (char*)oirpm_spec);
+      g_ptr_array_add (rpmbuild_argv, NULL);
+      int estatus;
+      GLNX_AUTO_PREFIX_ERROR ("Running rpmbuild", error);
+      if (!g_spawn_sync (oirpm_tmpd.path, (char**)rpmbuild_argv->pdata, NULL, G_SPAWN_SEARCH_PATH,
+                         NULL, NULL, NULL, NULL, &estatus, error))
+        return FALSE;
+      if (!g_spawn_check_exit_status (estatus, error))
+        return FALSE;
+    }
+  else
+    {
+      g_print ("Wrote: %s\n", oirpm_tmpd.path);
+      glnx_tmpdir_unset (&oirpm_tmpd);
+    }
+
+  return TRUE;
+}
+
+int
+rpmostree_compose_builtin_commit2jigdo (int             argc,
+                                        char          **argv,
+                                        RpmOstreeCommandInvocation *invocation,
+                                        GCancellable   *cancellable,
+                                        GError        **error)
+{
+  g_autoptr(GOptionContext) context = g_option_context_new ("COMMIT OIRPM-SPEC");
```

Comment:
Well, there are things like `%description`...I dunno.   Maybe support providing one but if not we autogenerate it?

Reply:
As far as `Requires`; I decided not to do this initially; the rationale is that we can easily get *other* synthetic `Requires` that RPM injects like `rpmlib(PayloadIsXz) <= 5.2-1` and so we'd have to learn to ignore those.  Another reason down the line is that I want the "jigdo set" to actually be metadata in the ostree commit, so it can be covered by that GPG signature.